control-plane-agent/docs/MCM-CONTROL-PLANE-API-DEFINITION.md

```diff
@@ -0,0 +1,302 @@
+# MCM Control Plane REST API Definition (**Control Plane API**)
+
+* Mesh Agent runs a server to handle **Control Plane API** requests.
+* **Control Plane API** is exposed to any external network configuration tools.
+* **Control Plane API** is a REST API based on HTTP 1.1.
+
+## List Media Proxies
+```
+GET /media-proxy
+```
+Query parameters
+```
+status - if 'true', include Media Proxy status
+config - if 'true', include Media Proxy configuration
+```
+**Status**: `200`
+```
+Content-Type: application/json
+
+{
+    "mediaProxy": [
+        {
+            "id": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
+            "status": {
+                "healthy": true,
+                "startedAt": 2024-10-17T17:24:53Z,
+                "connectionNum": 1
+            },
+        },
+        {
+            "id": "XXXXXXXX-YYYY-XXXX-XXXX-XXXXXXXXXXXX"
+            "status": {
+                "healthy": true,
+                "startedAt": 2024-10-17T17:24:54Z,
+                "connectionNum": 2
+            }
+        }
+    ]
+}
+```
+
+## Get Media Proxy Status
+```
+GET /media-proxy/{id}
+```
+Query parameters
+```
+config - if 'true', include Media Proxy configuration
+```
+**Status**: `200`
+```
+Content-Type: application/json
+
+{
+    "status": {
+        "healthy": true,
+        "startedAt": 2024-10-17T17:24:53Z,
+        "connectionNum": 1
+    },
+    "config": {
+        "ipAddr": "192.168.95.1"
+    }
+}
+```
+
+## List Connections
+```
+GET /connection
+```
+Query parameters
+```
+status - if 'true', include connection status
+config - if 'true', include connection configuration
+```
+**Status**: `200`
+```
+Content-Type: application/json
+
+{
+    "connection": [
+        {
+            "id": "XXXXXXXX-XXXX-YYYY-XXXX-XXXXXXXXXXXX",
+            "status": {
+                "createdAt": 2024-10-17T17:24:53Z,
+                "buffersSent": 301
+            }
+        }
+    ]
+}
+```
+
+## Get Connection Status
+```
+GET /connection/{id}
+```
+Query parameters
+```
+config - if 'true', include connection configuration
+```
+**Status**: `200`
+```
+Content-Type: application/json
+
+{
+    "status": {
+        "createdAt": 2024-10-17T17:24:53Z,
+        "buffersSent": 300
+    },
+    "config": {
+        "kind": "sender",
+        "connType": "ST2110",
+        "conn": {
+            "remoteIpAddr": "192.168.95.2",
+            "remotePort": 9001,
+        },
+        "payloadType": "video",
+        "payload": {
```

Comment:
Shouldn't this output look like an SDK json proposal (without connType/payloadType keys)?
```
 "connection": {
    "st2110": {
      "transport": "st2110-20" | "st2110-22" | "st2110-30" | "...",
      "remoteIpAddr": "192.168.95.2",
      "remotePort": "9002",
      "pacing": "narrow",
      "payloadType": 112
    },
},
  "payload": {
    "video": {
      "width": 1920,
      "height": 1080,
      "fps": 60.0,
      "pixelFormat": "yuv422p10le" | "..."
    }
}
```

Reply:
Agree. Let me mark this document as draft.

Reply:
Updated in c4aa2321fc5e57d1e9e13a90546aee334f7d0b2b